src/pins/gen.rs

```diff
@@ -79,7 +88,7 @@ macro_rules! pins {
             pub(crate) fn new() -> Self {
                 Self {
                     $(
-                        $field: Token($type(()), PhantomData),
+                        $field: Token(&$type(()), PhantomData),
```

Comment:
See comment on `Token` below. If my suggestion works out, we can switch to this:

```suggestion
                        $field: Token(PhantomData, PhantomData),
```